lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -6,19 +6,36 @@ class JsonApi < Adapter
         autoload :PaginationLinks
         autoload :FragmentCache
 
-        def initialize(serializer, options = {})
-          super
-          @included = ActiveModel::Serializer::Utils.include_args_to_hash(instance_options[:include])
-          fields = options.delete(:fields)
-          if fields
-            @fieldset = ActiveModel::Serializer::Fieldset.new(fields, serializer.json_key)
-          else
-            @fieldset = options[:fieldset]
+        module ApiObjects
+          # Make JSON API top-level jsonapi member opt-in
+          # ref: http://jsonapi.org/format/#document-top-level
+          ActiveModel::Serializer.config.jsonapi_include_toplevel_member = false
+          ActiveModel::Serializer.config.jsonapi_version = '1.0'
+          ActiveModel::Serializer.config.jsonapi_toplevel_meta = {}
+          module JsonApi
+            module_function
+
+            def include_member?
+              ActiveModel::Serializer.config.jsonapi_include_toplevel_member
+            end
+
+            def add!(document)
+              object = {
+               jsonapi: {
+                 version: ActiveModel::Serializer.config.jsonapi_version,
+                 meta: ActiveModel::Serializer.config.jsonapi_toplevel_meta
+               }
+              }
+              object[:jsonapi].reject! { |_, v| v.blank? }
+              document.merge!(object)
+            end
           end
         end
 
         def serializable_hash(options = nil)
           options ||= {}
+          self.hash = {}
```

Comment:
No I mean, why work on a `self.hash`/`@hash` instead of building an object in the method and returning it?